adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PathCache.java

```diff
@@ -115,12 +123,19 @@ private PathConfig removeIfExpired(CacheEntry cached) {
             return null;
         }
 
+        PathConfig config = cached.value();
+
         if (cached.isExpired()) {
             remove(cached.key());
+            
+            if (config != null && config.getPath() != null) {
+                // also remove from pre-defined paths map so that changes on the server are properly reflected
+                paths.remove(config.getPath());
```

Comment:
Hello @pedroigor, is it correct, that this way all paths defined only in enforcerConfig are removed after cache expiration? I tried to play with `ServletPolicyEnforcerTest` and changed cache [lifespan to 5ms](https://github.com/mhajas/keycloak/commit/c5b8a4dbb7a6f9bfd96240cdd5096ebdcf004252#diff-43416b6ffedd5e3ae06f87ea06cb6e30R13).

Now I tried to run `ServletPolicyEnforcerTest#testPattern1` and it fails during the first login. I described behavior in comments [here](https://github.com/mhajas/keycloak/commit/c5b8a4dbb7a6f9bfd96240cdd5096ebdcf004252#diff-8d6aca4c87112cf4206ec60bb213743aR575).

Reply:
Yes, the idea is to remove from any reference to paths, including those defined in the deployment configuration. The reason is that without it, you'll never get the application reflecting the latest changes made to resources/paths from the server. I'll look your comments.

Reply:
@mhajas I see your point in that comment. And the reason is that `lazy-load-paths` is set to false so the resource will never be fetched again.

I guess we should only remove if that flag is marked as true so that the application can fetch again resources from server. 

But your comment raises another question if you have paths with a specific configuration in your deployment configuration, they'll also be lost ... Let me create some tests based on your findings.

Reply:
@mhajas Could you check now ? I also updated tests.